sdk/apimanagement/api-management-custom-widgets-tools/src/node/CustomWidgetBlobService.ts

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { BlobServiceClient, BlockBlobUploadResponse } from "@azure/storage-blob";
+import { buildBlobConfigSrc, buildBlobDataSrc } from "../paths";
+import mime from "mime";
+
+type TConfig = Record<string, unknown>;
+
+/**
+ * A service wrapping ContainerClient class to simplify blob handling
+ */
+class CustomWidgetBlobService {
+  containerClient;
+  name;
+  pathWidget;
+  pathConfig;
+
+  /**
+   * @param blobStorageUrl - blob storage SAS URL
+   * @param name - name of the custom widget to be taken care of
+   */
+  constructor(blobStorageUrl: string, name: string) {
+    const container = "content";
+    const blobServiceClient = new BlobServiceClient(blobStorageUrl.replace(`/${container}`, ""));
+    this.containerClient = blobServiceClient.getContainerClient(container);
+    this.name = name;
+    this.pathWidget = buildBlobDataSrc(name);
+    this.pathConfig = buildBlobConfigSrc(name);
+  }
+
+  fileName(path: string): string | undefined {
+    return path.split("/").pop();
+  }
+
+  async blobUpload(absolutePath: string, content: Buffer): Promise<BlockBlobUploadResponse> {
+    const fileName = this.fileName(absolutePath);
+    if (!fileName) throw new Error("a fileName was not found in the absolutePath");
+    return this.containerClient.getBlockBlobClient(absolutePath).upload(content, content.length, {
+      blobHTTPHeaders: { blobContentType: mime.getType(fileName) || "application/octet-stream" },
+    });
+  }
+
+  async jsonUpload(
+    absolutePath: string,
+    json: Record<string, unknown>
```

Comment:
I would just call this `content` instead of JSON.

Reply:
there is a generic `blobUpload`, I wanted to make clear it takes a JSON object